File: example/src/main/java/ru/virgil/example/order/BuyingOrderRepository.java
package ru.virgil.example.order;

import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import ru.virgil.example.user.UserDetails;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Repository
public interface BuyingOrderRepository extends JpaRepository<BuyingOrder, UUID> {

    List<BuyingOrder> findAllByOwner(UserDetails owner, Pageable pageable);

    Optional<BuyingOrder> findByOwnerAndUuid(UserDetails owner, UUID uuid);

    long countAllByOwner(UserDetails owner);

}

File: example/src/main/java/ru/virgil/example/order/BuyingOrderRepository.kt
package ru.virgil.example.order

import org.springframework.data.domain.Pageable
import org.springframework.stereotype.Repository
import ru.virgil.example.system.entity.OwnedEntityRepository
import ru.virgil.example.truck.Truck
import ru.virgil.example.user.UserDetails
import java.util.*

@Repository
interface BuyingOrderRepository : OwnedEntityRepository<BuyingOrder> {

    fun findAllByOwner(owner: UserDetails, pageable: Pageable): List<BuyingOrder>
    fun findByOwnerAndUuid(owner: UserDetails, uuid: UUID): Optional<BuyingOrder>
    fun countAllByOwner(owner: UserDetails): Long
    fun findAllByTruck(truck: Truck, pageable: Pageable): List<BuyingOrder>
}
